server/jetstream_benchmark_test.go

```diff
@@ -1295,6 +1295,187 @@ func BenchmarkJetStreamObjStore(b *testing.B) {
 	}
 }
 
+func BenchmarkJetStreamFanIn(b *testing.B) {
+	const (
+		subject     = "test-subject"
+		streamName  = "test-stream"
+		clusterSize = 3
+		replicas    = 3
+	)
+
+	type BenchPublisher struct {
+		// nats connection for this publisher
+		conn *nats.Conn
+		// jetstream context
+		js nats.JetStreamContext
+		// number of publishing errors encountered
+		publishErrors int
+		// quit channel which will terminate publishing
+		quitCh chan bool
+	}
+
+	messageSizeCases := []int64{
+		100,        // 100B
+		1024,       // 1KiB
+		10240,      // 10KiB
+		512 * 1024, // 512KiB
+	}
+	numPubsCases := []int{
+		3,
+		5,
+		10,
+	}
+
+	fanIn := func(b *testing.B, numPubs int, messageSize int64) {
+
+		cl, ls, shutdown, nc, js := startJSClusterAndConnect(b, clusterSize)
+		defer shutdown()
+		defer nc.Close()
+
+		clientUrl := ls.ClientURL()
+
+		_, err := js.AddStream(&nats.StreamConfig{
+			Name:     streamName,
+			Subjects: []string{subject},
+			Replicas: replicas,
+		})
+		if err != nil {
+			b.Fatal(err)
+		}
+		defer js.DeleteStream(streamName)
+
+		// If replicated resource, connect to stream leader for lower variability
+		if replicas > 1 {
+			nc.Close()
+			clientUrl = cl.streamLeader("$G", streamName).ClientURL()
+			nc, _ = jsClientConnectURL(b, clientUrl)
+			defer nc.Close()
+		}
+
+		// random bytes as payload
+		messageData := make([]byte, messageSize)
+		if _, err := rand.Read(messageData); err != nil {
+			b.Fatal(err)
+		}
+
+		// waits for all publishers sub-routines and for main thread to be ready
+		var publishersReadyWg sync.WaitGroup
+		publishersReadyWg.Add(numPubs + 1)
+
+		// create N publishers
+		publishers := func() []BenchPublisher {
```

Comment:
Here you are just creating 10 connections and 10 channels. 
You don't need a function and you don't need a waitgroup.. 
Just create 10 publisher in a for loop?